src/test/java/com/wikia/webdriver/testcases/discussions/ReportingPostTests.java

```diff
@@ -0,0 +1,620 @@
+          "moderatorOnDesktopCanApproveReportedPostOnPostDetailsPage"})
+  @Execute(asUser = User.USER_3)
+  @InBrowser(browser = Browser.FIREFOX, browserSize = DESKTOP_RESOLUTION)
+  public void userOnDesktopCanReportApprovedPostOnPostDetailsPage() {
+    final String postId = postData.get(POST_ON_DESKTOP_ON_POST_DETAILS).getId();
+
+    PostDetailsPage page = new PostDetailsPage().open(postId);
+
+    assertThatPostReportedByOtherUserDoesNotHaveReportedIndicatorAndCanBeReportedByCurrentUser(page, postId);
+  }
+
+  // Discussions moderator on desktop
+
+  @Test(groups = "discussions-loggedInDiscussionsModeratorDesktopReporting",
+      dependsOnMethods = {
+          "userOnDesktopCanReportPostOnPostDetailsPage",
+          "anonUserOnDesktopCanNotSeeReportedPostOnPostDetailsPage",
+          "userOnDesktopCannotSeeReportedIndicatorOnPostsReportedByAnotherUserOnPostDetailsPageAndCanReportThatPost"})
+  @Execute(asUser = User.DISCUSSIONS_MODERATOR)
+  @InBrowser(browser = Browser.FIREFOX, browserSize = DESKTOP_RESOLUTION)
+  public void moderatorOnDesktopCanApproveReportedPostOnPostDetailsPage() {
+    final String postId = postData.get(POST_ON_DESKTOP_ON_POST_DETAILS).getId();
+
+    PostDetailsPage page = new PostDetailsPage().open(postId);
+
+    assertThatDiscussionsModeratorCanSeeAndApproveReportedPost(page, postId);
+  }
+
+  @Test(groups = "discussions-loggedInDiscussionsModeratorDesktopReporting",
+      dependsOnMethods = {
+          "userOnDesktopCanReportPostOnPostDetailsPage",
+          "anonUserOnDesktopCanNotSeeReportedPostOnPostDetailsPage",
+          "userOnDesktopCannotSeeReportedIndicatorOnPostsReportedByAnotherUserOnPostDetailsPageAndCanReportThatPost",
+          "moderatorOnDesktopCanApproveReportedPostOnPostDetailsPage",
+          "userOnDesktopCannotReReportPostOnPostDetailsPage",
+          "userOnDesktopCanReportApprovedPostOnPostDetailsPage"})
+  @Execute(asUser = User.DISCUSSIONS_MODERATOR)
+  @InBrowser(browser = Browser.FIREFOX, browserSize = DESKTOP_RESOLUTION)
+  public void moderatorOnDesktopCanDeleteReportedPostOnPostDetailsPage() {
+    final String postId = postData.get(POST_ON_DESKTOP_ON_POST_DETAILS).getId();
+
+    PostDetailsPage page = new PostDetailsPage().open(postId);
+
+    assertThatDiscussionsModeratorCanSeeAndRemoveReportedPost(page, postId);
+  }
+
+  private void assertThatReportPostOptionIsNotAvailableOn(final PageWithPosts page) {
+    assertThatReportPostOptionIsNotAvailableFor(page.getPost().findNewestPost());
+  }
+
+  private void assertThatNoPostHasReportedIndicator(final PageWithPosts pageWithPosts) {
+    Assertion.assertTrue(pageWithPosts.getPost().getReportedPosts().isEmpty(),
+        "Anonymous user should not see reported indicator.");
+  }
+
+  private PostEntity createAndGetNewPost(final PostsListPage postsListPage, final PostsCreator postsCreator) {
+    PostEntity.Data postData = postsCreator.click()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename "postData" which hides the field declared at line 41. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AHiddenFieldCheck)